cpp/src/fil/treelite_import.cu

```diff
@@ -329,96 +329,23 @@ conversion_state<fil_node_t> tl2fil_inner_node(int fil_left_child,
-  }
+  fil_node_t node({}, split, feature_id, default_left, false, is_categorical, fil_left_child);
   return conversion_state<fil_node_t>{node, tl_left, tl_right};
 }
 
-template <typename T, typename L>
-void node2fil_dense(std::vector<dense_node>* pnodes,
-                    int root,
-                    int cur,
-                    const tl::Tree<T, L>& tree,
-                    int node_id,
-                    const forest_params_t& forest_params,
-                    std::vector<float>* vector_leaf,
-                    std::size_t* leaf_counter,
-                    cat_sets_owner* cat_sets,
-                    std::size_t* bit_pool_offset)
-{
-  if (tree.IsLeaf(node_id)) {
-    (*pnodes)[root + cur] = dense_node({}, {}, 0, false, true, false);
-    tl2fil_leaf_payload(
-      &(*pnodes)[root + cur], root + cur, tree, node_id, forest_params, vector_leaf, leaf_counter);
-    return;
-  }
-
-  // inner node
-  int left = 2 * cur + 1;
-  conversion_state<dense_node> cs =
-    tl2fil_inner_node<dense_node>(left, tree, node_id, forest_params, cat_sets, bit_pool_offset);
-  (*pnodes)[root + cur] = cs.node;
-  node2fil_dense(pnodes,
-                 root,
-                 left,
-                 tree,
-                 cs.tl_left,
-                 forest_params,
-                 vector_leaf,
-                 leaf_counter,
-                 cat_sets,
-                 bit_pool_offset);
-  node2fil_dense(pnodes,
-                 root,
-                 left + 1,
-                 tree,
-                 cs.tl_right,
-                 forest_params,
-                 vector_leaf,
-                 leaf_counter,
-                 cat_sets,
-                 bit_pool_offset);
-}
-
-template <typename T, typename L>
-void tree2fil_dense(std::vector<dense_node>* pnodes,
-                    int root,
-                    const tl::Tree<T, L>& tree,
-                    std::size_t tree_idx,
-                    const forest_params_t& forest_params,
-                    std::vector<float>* vector_leaf,
-                    std::size_t* leaf_counter,
-                    cat_sets_owner* cat_sets)
-{
-  node2fil_dense(pnodes,
-                 root,
-                 0,
-                 tree,
-                 tree_root(tree),
-                 forest_params,
-                 vector_leaf,
-                 leaf_counter,
-                 cat_sets,
-                 &cat_sets->bit_pool_offsets[tree_idx]);
-}
-
 template <typename fil_node_t, typename T, typename L>
-int tree2fil_sparse(std::vector<fil_node_t>& nodes,
-                    int root,
-                    const tl::Tree<T, L>& tree,
-                    std::size_t tree_idx,
-                    const forest_params_t& forest_params,
-                    std::vector<float>* vector_leaf,
-                    std::size_t* leaf_counter,
-                    cat_sets_owner* cat_sets)
+int tree2fil(std::vector<fil_node_t>& nodes,
```

Comment:
I wonder if this would better be a method of `tl2fil_t` instead.

Feel free to do it in another pull request, though.